api.planx.uk/webhooks/hardDeleteSessions.test.ts

```diff
@@ -0,0 +1,67 @@
+import supertest from "supertest";
+import app from "../server";
+import { queryMock } from "../tests/graphqlQueryMock";
+
+const { post } = supertest(app);
+const ENDPOINT = "/webhooks/hasura/delete-expired-sessions";
+
+describe("Delete expired sessions webhook", () => {
+  afterEach(() => queryMock.reset());
+
+  it("fails without correct authentication", () => {
+    post(ENDPOINT)
+      .expect(401)
+      .then((response) => {
+        expect(response.body).toEqual({
+          error: "Unauthorised",
+        });
+      });
+  });
+
+  it("returns a positive response with authentication", async () => {
+    queryMock.mockQuery({
+      name: "HardDeleteExpiredSessions",
+      data: {
+        delete_lowcal_sessions: {
+          returning: [
+            { id: "abc123" },
+            { id: "xyz789" },
+          ]
+        }
+      },
+      ignoreThesePropertiesInVariables: [
+        "oneWeekAgo"
+      ]
```

Comment:
I spent waaaay to long messing with Jest's fake timers trying to get a usable date for this mock query.

Things I learned - 
 - `QueryMock` has this awesome property `ignoreThesePropertiesInVariables` that allowed me to totally bypass the problem ([docs](https://github.com/zth/graphql-query-test-mock/blob/master/README.md#how-to-i-handle-unstable-variables-like-date-or-random-things-that-change-for-every-query))
 - Jest's fake timers do not work with Express. I never got to the _exact_ reason for this but it's to do with how Express uses the native timer (so Jest's implementation gets ignored?)
 - I was able to get around this using [`@sinon-js/fake-timers`](https://github.com/sinonjs/fake-timers). Something like this - 
 
 
 ```ts
 let clock: FakeTimers.InstalledClock;
  
beforeAll(() => clock = FakeTimers.install());
afterEach(() => clock.uninstall());

test("something", () => {
    clock.setSystemTime(new Date(2022, 10, 8))
    // Test stuff....
});
```

In this instance `ignoreThesePropertiesInVariables` worked great as the timing logic doesn't need to be tested, I just needed values for my mock query.